monailabel/tasks/infer/basic_infer.py

```diff
@@ -273,6 +273,7 @@ def __call__(
         ) if self.path and isinstance(self.path, list) else self.path
 
         # device
+        req.update({"device": device_list()})
         device = req.get("device", "cuda")
         device = device if isinstance(device, str) else device[0]
         if device.startswith("cuda") and not torch.cuda.is_available():
```

Comment:
some changes needed here i believe.. because there is specific cuda check happening..

also `req.update({"device": device_list()})` is not correct in some context..

Reply:
Then, line 277 also needs to be updated as it is looking for "cuda".

Reply:
Hi @japarada , could you resolve issue so that we can have basic tests? 
` req.update({"device": device_list()})` is not in a right place currently. 

Thank you!